src/main/java/com/percero/amqp/PerceroAgentListener.java

```diff
@@ -36,6 +36,11 @@
  */
 @Component("perceroAgentListener")
 public class PerceroAgentListener implements MessageListener {
+	
+	private static PerceroAgentListener perceroAgentListener = null;
```

Comment:
When we're building a framework, yes.  Because we want our code to be as
'plain' as possible so that outside devs can use it however they like. We
need to continue to use DI patterns but do it more manually.

On Thu, Sep 24, 2015 at 1:39 PM, Collin Brown <notifications@github.com>
wrote:

> In src/main/java/com/percero/amqp/PerceroAgentListener.java
> <https://github.com/ActiveStack/syncengine/pull/7#discussion_r40369901>:
>
> > @@ -36,6 +36,11 @@
> >   */
> >  @Component("perceroAgentListener")
> >  public class PerceroAgentListener implements MessageListener {
> > +	
> > +	private static PerceroAgentListener perceroAgentListener = null;
>
> So you are in favor of singleton over Spring?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ActiveStack/syncengine/pull/7/files#r40369901>.
>



-- 
Jonathan